portlayer/exec/container.go

```diff
@@ -80,44 +86,78 @@ func (c *Container) newHandle() *Handle {
 	return newHandle(c)
 }
 
-func (c *Container) Commit(ctx context.Context, sess *session.Session, s *types.VirtualMachineConfigSpec) error {
+func (c *Container) cacheExecConfig(ec *metadata.ExecutorConfig) {
 	c.Lock()
 	defer c.Unlock()
 
-	if c.vm != nil {
-		// reconfigure
-		_, err := tasks.WaitForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
-			return c.vm.Reconfigure(ctx, *s)
-		})
+	c.ExecConfig = ec
+}
 
-		if err != nil {
-			return err
-		}
-	} else {
-		// Find the Virtual Machine folder that we use
-		folders, err := sess.Datacenter.Folders(ctx)
-		if err != nil {
-			return err
+func (c *Container) Commit(ctx context.Context, sess *session.Session, h *Handle) error {
+	c.Lock()
```

Comment:
I remember @caglar10ur and @fdawg4l had a good reason for naming your mutexes inside of structs rather than embedding them. It may not apply here, but I CC'd them just in case. Maybe one of them can refresh my memory.